webapi/src/main/java/org/sakaiproject/webapi/controllers/PermissionsController.java

```diff
@@ -121,7 +122,7 @@ public Map<String, Object> getPermissions(@PathVariable String siteId, @PathVari
 
                 on.put(role.getId(), filteredFunctions);
 
-                if (overrideRole != null) {
+                if (overrideRole != null && !"content".equalsIgnoreCase(tool)) {
```

Comment:
Adrian we reviewed this in the core call and didn't quite understand the authz override, could you help clarify?

Reply:
what no one could understand on the core call: what is the use case for locking the permissions so that instructor cannot modify them? does this PR look correct to you?

Reply:
Because top level folder permissions are supposed to dominate. I did it on purpose base on Sean Horner's description of the use case.

Reply:
Sorry this isn't making sense to me. The use case that we've supported for a dozen years is creating a sub-folder inside Resources that has increased permissions (e.g., students can add/edit files). That seems to no longer be possible. 

I see Sean advocating for disabling checkboxes in https://sakaiproject.atlassian.net/browse/SAK-50984, but I think this hasn't been fully thought through.

The root of the problem seems to be that the sub-folder realm permissions are additive on top of the parent folder permissions? E.g., if top parent has content.delete for student role than it's impossible to remove that permission from a sub-folder?

Reply:
Yes. That's what Sean described to me and I assumed that was the way it was in the past. Tell me how it's supposed to be and I'll fix it. It will be an easy fix.

Reply:
1. Instructor can edit permissions at top-level and all sub-folders
2. Don't create a realm on the sub-folder until the save goes through (existing behavior)
3. sub-folder permissions rule that folder (not additive permissions with parent folders)
4. permissions are never disabled/locked

Reply:
Ok, I'll sort that out now if you're ok with me doing it.

Reply:
of course! 👍

Reply:
The way Sean described how resources worked in 21 and maybe 22 does seem to read as though a switched on permission from the top level folder should dominate. That's why I disable those turned on permissions from the override authzgroup, stopping anyone from toggling them off. He was quite specific. Pre the PR I did on SAK-50984 I think things worked the way your describe - the authzgroups operating independently.

Reply:
But isn't Sean's whole "lock" proposal just cementing the top-level folder permissions instead of fixing the actual issue? The actual issue is that the instructor believes that the folder permissions will work exactly as displayed. the locking scheme was an attempt to somehow convey that to the user. but it's all bad instincts. the real fix is that the permissions for a sub-folder should work exactly as displayed (e.g., no "magic" permissions bubbling up from parent folders)!

Reply:
What did he mean by this?

1. Sakai 23 & 25 now allow instructors to revoke permissions granted in parent folders 
2. Sakai 23 & 25 no longer inherits permissions from parent folders

I just assumed he was correct about previous behaviour being broken by my switching to the permissions component. I assumed that disabling or locking the "on" permissions inherited from the top level folder reflected the way the permissions checks were happening in the content tool or service.

Reply:
Sakai 23 has changed a lot after the permissions helper was removed and permissions web component replaced it.

Here is the way Sakai 22 works today:

1. The root Resources tool permissions are the base-level permissions.
2. sub-folder permissions inherit from the parent folder permissions and from the tool permissions.
3. You cannot remove permissions from a folder that are defined in (1) or (2). these are locked/disabled
4. if a permission is granted on a parent folder, there is no way to remove it on a sub-folder.


IDK, this isn't what i assumed. do you want more people to weight in?

Reply:
I think we need to codify how this is supposed/expected to work so it probably makes sense for others to chime in. I'm worried that if I change things again somebody else will complain that I've broken it. You know how it goes.

Reply:
I thought it had gone all too well when I switched content to the web component ...

Reply:
i'm fine with implementing this exactly as it behaves on Sakai 22. but i think it's ultimately all wrong ;) but at least it would be consistent